server/container/guice/guice-common/src/main/java/org/apache/james/modules/CommonServicesModule.java

```diff
@@ -65,6 +75,37 @@ protected void configure() {
         bind(serverProbe).in(Singleton.class);
     }
 
+    private void listenToConfigurables() {
+        Configurables configurables = new Configurables();
+        bind(Configurables.class).toInstance(configurables);
+
+        bindListener(new SubclassesOf(Configurable.class), new TypeListener() {
+
+            @Override
+            public <I> void hear(TypeLiteral<I> type, TypeEncounter<I> encounter) {
+                MembersInjector<? super I> membersInjector = x -> {
+                    Class<? super I> rawType = type.getRawType();
+                    configurables.add(rawType.asSubclass(Configurable.class));
+                };
+                encounter.register(membersInjector);
+            }
+        });
+    }
+
+    private static class SubclassesOf extends AbstractMatcher<TypeLiteral<?>> implements Serializable {
```

Comment:
No we can't use it as it's a `Matcher<Class>` and we need a `Matcher<TypeLiteral>`